app/questionnaire/questionnaire_schema.py

```diff
@@ -439,28 +487,37 @@ def get_relationship_collectors_by_list_name(self, list_name: str):
                 if block["for_list"] == list_name
             ]
 
-    def get_unrelated_block_no_answer_values(self, unrelated_answer_id):
-        return [
-            option["value"]
-            for unrelated_answer in self.get_answers_by_answer_id(unrelated_answer_id)
-            for option in unrelated_answer["options"]
-            if option.get("action", {}).get("type") == "AddUnrelatedRelationships"
-        ]
+    def get_unrelated_block_no_answer_values(
+        self, unrelated_answer_id: str
+    ) -> Optional[list[str]]:
+        if unrelated_answers := self.get_answers_by_answer_id(unrelated_answer_id):
+            return [
+                option["value"]
+                for unrelated_answer in unrelated_answers
+                for option in unrelated_answer["options"]
+                if option.get("action", {}).get("type") == "AddUnrelatedRelationships"
+            ]
 
     @staticmethod
-    def get_single_string_value(schema_object):
+    def get_single_string_value(schema_object: Union[Mapping, str]) -> str:
         """
         Resolves an identifying string value for the schema_object. If text_plural the `other` form is returned.
         :return: string value
         """
-        if isinstance(schema_object, dict):
-            if "text_plural" in schema_object:
-                return schema_object["text_plural"]["forms"]["other"]
-            return schema_object["text"]
-        return schema_object
+        if isinstance(schema_object, str):
+            return schema_object
+
+        if "text_plural" in schema_object:
+            plural_placeholder_string: str = schema_object["text_plural"]["forms"][
+                "other"
+            ]
+            return plural_placeholder_string
+
+        placeholder_string: str = schema_object["text"]
+        return placeholder_string
 
     @staticmethod
-    def get_all_questions_for_block(block):
+    def get_all_questions_for_block(block: Mapping) -> list[ImmutableDict]:
         all_questions = []
```

Comment:
We should specify types for empty collections?

Reply:
Do you mean the return type? Is that not covered by the existing type?

Reply:
Good spot, I have added types for all the variables initialised with an empty collection.